File: truss-compiler/src/main/java/truss/compiler/TrussArguments.java
package truss.compiler;

import truss.compiler.support.ArgumentFlag;
import truss.compiler.support.ArgumentMultiple;
import truss.compiler.support.ArgumentOption;
import truss.compiler.support.Arguments;

import java.util.List;

public class TrussArguments extends Arguments {
    private final ArgumentFlag noCorLib = new ArgumentFlag("--no-corlib", "Do not link corlib.dar");
    private final ArgumentOption output = new ArgumentOption("-o", "Output file name", true);
    private final ArgumentMultiple link = new ArgumentMultiple("-l", "DTL libraries to link against", false);

    public TrussArguments(String[] args) throws ArgumentException {
        addArgument(noCorLib);
        addArgument(output);
        addArgument(link);

        parse(args);
    }

    public boolean isNoCorLib() {
        return noCorLib.isProvided();
    }

    public String getOutput() {
        return output.getValue();
    }

    public List<String> getLink() {
        return link.getValues();
    }

    public List<String> getFiles() {
        return getRemaining();
    }
}

File: Truss.Compiler/TrussArguments.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Truss.Compiler.Support;

namespace Truss.Compiler {
    public class TrussArguments : Arguments {
        private readonly ArgumentFlag _noCorLib = new ArgumentFlag("--no-corlib", "Do not link corlib.dar");
        private readonly ArgumentOption _output = new ArgumentOption("-o", "Output file name", true);
        private readonly ArgumentMultiple _link = new ArgumentMultiple("-l", "DTL libraries to link against", false);

        public TrussArguments(string[] args) {
            AddArgument(_noCorLib);
            AddArgument(_output);
            AddArgument(_link);

            Parse(args);
        }

        public bool NoCorLib {
            get { return _noCorLib.IsProvided; }
        }

        public string Output {
            get { return _output.Value; }
        }

        public List<string> Link {
            get { return _link.Values; }
        }

        public List<string> Files {
            get { return Remaining; }
        }
    }
}
